dessia_common/excel_reader.py

```diff
@@ -0,0 +1,503 @@
+
+            return actual_value
+
+        if isinstance(values, list) and (not any(type_ in cell.value for type_ in container)):
+            actual_value = values[0]
+        else:
+            actual_value = values
+
+        return actual_value
+
+    @staticmethod
+    def get_attributes_and_types(obj_class):
+        """
+        Retrieve attributes and their corresponding types from the __init__ method of a class.
+
+        Args:
+        - obj_class: The class object.
+
+        Returns:
+        - dict: A dictionary containing attributes and their respective types defined in the class constructor.
+        """
+        init_signature = inspect.signature(obj_class.__init__)
+        init_parameters = init_signature.parameters
+
+        attribute_info = {}
+
+        for attr_name, param in init_parameters.items():
+            if attr_name != 'self':
+                attr_info = {
+                    'type': param.annotation if param.annotation != param.empty else None,
+                    'default_value': param.default if param.default != param.empty else "empty"
+                }
+                attribute_info[attr_name] = attr_info
+
+        return attribute_info
+
+    @staticmethod
+    def get_data(value, attributes, init_attributes):
+        """
+        Extracts relevant data based on attributes and initial attributes for object instantiation.
+
+        Args:
+        - value (list): List containing values corresponding to attributes.
+        - attributes (list): List of attributes related to the object.
+        - init_attributes (list): List of initial attributes for object instantiation.
+
+        Returns:
+        - dict: Dictionary containing extracted object data based on matching attributes and initial attributes.
+        """
+        detect_missing_attribute, attribute = missed_attribute(attributes, init_attributes)
+        if not detect_missing_attribute:
+            raise ValueError(f"Missing attribute '{attribute}' doesn't have a default value.")
+
+        object_data = {}
+        for i, attribute in enumerate(attributes):
+            if attribute in init_attributes:
+                if isinstance(value[i], str):
+                    try:
+                        object_data[attribute] = literal_eval(value[i])
+                    except (ValueError, SyntaxError):
+                        object_data[attribute] = value[i]
+                else:
+                    object_data[attribute] = value[i]
+        return object_data
+
+    def create_object(self, obj_class, value_set, attributes, initial_attributes):
+        """Create an object of a specified class using provided data."""
+        object_data = self.get_data(value_set, attributes, initial_attributes)
+        object_ = obj_class(**object_data)
+        if not object_.name:
+            object_.name = ""
+
+        return object_
+
+    def instantiate_main_object(self, instantiated_objects, class_info, attributes, key, values):
```

Comment:
You should mutualize these attributes by using ExcelDataExtract. Consider this : 

```py
class ExcelDataExtract:
    """Data extracted from sheet of Excel file."""

    def __init__(self, class_info: Tuple[str, str], attributes: List[str], datas):
        self.class_info = class_info
        self.attributes = attributes
        self.datas = datas

    @property
    def object_class(self) -> str:
        return f"{self.class_info[0]}.{self.class_info[1]}"

    @property
    def class_(self):
        return get_python_class_from_class_name(self.object_class)


def instantiate_main_object(self, instantiated_objects, extracted_data: ExcelDataExtract, key):
    """
    Instantiate main objects based on provided values and attributes.
    Args:
    - instantiated_objects (dict): Dictionary containing instantiated objects.
    - key (str): Key to identify the instantiated objects.
    - values (list): List containing module name, class name, and attributes.
    Returns:
    - dict: Updated dictionary of instantiated objects.
    """
    initial_attributes = self.get_attributes_and_types(extracted_data.class_)
    objects = []

    value_set = list(extracted_data.datas.values())[0]
    for index, value in enumerate(value_set):
        if isinstance(value, openpyxl.cell.cell.Cell):
            replaced_value = instantiated_objects[self.get_location(value)[0]]
            value_set[index] = self.update_attribute_values(value, replaced_value)

    objects.append(self.create_object(extracted_data, value_set, initial_attributes))

    instantiated_objects[key] = objects
    return instantiated_objects
```